show/main.py

```diff
@@ -349,12 +351,95 @@ def mac(vlan, port):
 #
 
 # This group houses IP (i.e., IPv4) commands and subgroups
-@cli.group()
+@cli.group(cls=AliasedGroup, default_if_no_args=False)
 def ip():
     """Show IP (IPv4) commands"""
     pass
 
 
+#
+# get_if_admin_state
+#
+# Given an interface name, return its admin state reported by the kernel.
+#
+def get_if_admin_state(iface):
+    admin_file = "/sys/class/net/{0}/flags"
+
+    try:
+        state_file = open(admin_file.format(iface), "r")
+    except IOError as e:
+        print "Error: unable to open file: %s" % str(e)
+        return "error"
+
+    content = state_file.readline().rstrip()
+    flags = int(content, 16)
+
+    if flags & 0x1:
+        return "up"
+    else:
+        return "down"
+
+
+#
+# get_if_oper_state
+#
+# Given an interface name, return its oper state reported by the kernel.
+#
+def get_if_oper_state(iface):
+    oper_file = "/sys/class/net/{0}/carrier"
+
+    try:
+        state_file = open(oper_file.format(iface), "r")
+    except IOError as e:
+        print "Error: unable to open file: %s" % str(e)
+        return "error"
+
+    oper_state = state_file.readline().rstrip()
+    if oper_state == "1":
+        return "up"
+    else:
+        return "down"
+
+
+#
+# 'show ip interfaces' command
+#
+# Display all interfaces with an IPv4 address and their admin/oper states.
+# Addresses from all scopes are included. Interfaces with no addresses are
+# excluded.
+#
+@ip.command()
+def interfaces():
+    """Show interfaces IPv4 address"""
+    header = ['Interface', 'IPv4 address/mask', 'Admin/Oper']
+    data = []
+
+    interfaces = natsorted(netifaces.interfaces())
+
+    for iface in interfaces:
+
+        ipaddresses = netifaces.ifaddresses(iface)
+
+        if netifaces.AF_INET in ipaddresses:
+
```

Comment:
Delete empty line after `if` for consistent style.

Reply:
Done.